cmd/aws-vpc-cni/main.go

```diff
@@ -384,14 +387,29 @@ func validateEnvVars() bool {
 	}
 
 	prefixDelegationEn := utils.GetBoolAsStringEnvVar(envEnPrefixDelegation, defaultEnPrefixDelegation)
-	warmIPTarget := utils.GetEnv(envWarmIPTarget, "0")
-	warmPrefixTarget := utils.GetEnv(envWarmPrefixTarget, "0")
-	minimumIPTarget := utils.GetEnv(envMinIPTarget, "0")
+	warmIPTarget, err, input := utils.GetIntFromStringEnvVar(envWarmIPTarget, defaultWarmIPTarget)
+	if err != nil {
+		log.Errorf("error when trying to get env WARM_IP_TARGET: %s; input is %v", err, input)
+		return false
+	}
+	warmPrefixTarget, err, input := utils.GetIntFromStringEnvVar(envWarmPrefixTarget, defaultWarmPrefixTarget)
+	if err != nil {
+		log.Errorf("error when trying to get env WARM_PREFIX_TARGET: %s; input is %v", err, input)
+		return false
+	}
+	minimumIPTarget, err, input := utils.GetIntFromStringEnvVar(envMinIPTarget, defaultMinIPTarget)
+	if err != nil {
+		log.Errorf("error when trying to get env MINIMUM_IP_TARGET: %s; input is %v", err, input)
+		return false
+	}
 
 	// Note that these string values should probably be cast to integers, but the comparison for values greater than 0 works either way
-	if prefixDelegationEn && (warmIPTarget <= "0" && warmPrefixTarget <= "0" && minimumIPTarget <= "0") {
+	if prefixDelegationEn && (warmIPTarget <= 0 && warmPrefixTarget <= 0 && minimumIPTarget <= 0) {
 		log.Errorf("Setting WARM_PREFIX_TARGET = 0 is not supported while WARM_IP_TARGET/MINIMUM_IP_TARGET is not set. Please configure either one of the WARM_{PREFIX/IP}_TARGET or MINIMUM_IP_TARGET env variables")
 		return false
+	} else if warmIPTarget <= 0 && minimumIPTarget > 0 {
```

Comment:
nit: I think this check should be: `if (minimumIPTarget is set and warmIPTarget is not set) { ... }`
And the way we can check if they are set is by using the `input` value returned from `GetIntFromStringEnvVar`

Reply:
Updated, thanks